src/classes/ExpectSObject.cls

```diff
@@ -0,0 +1,72 @@
+public class ExpectSObject extends Expect {
+    private final SObject expected;
+
+    public ExpectSObject(SObject expected) {
+        this.expected = expected;
+    }
+
+    protected override String getExpectedAsString() {
+        return String.valueOf(expected);
+    }
+
+    public ExpectSObject andIt { get { return this; } }
+
+    public ExpectSObject shouldEqual(SObject actual) {
+        shouldEqual(actual, expectedMessage(actual,''));
+        return this;
+    }
+
+    public ExpectSObject shouldEqual(SObject actual, String message) {
+        return assertSObject(expected === actual, message);
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual) {
+        return shouldNotEqual(actual, expectedMessage(actual,' not'));
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual, String message) {
+        return assertSObject(expected !== actual, message);
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual) {
+        return sObjectTypeEquals(actual, expectedMessage(actual.getSObjectType(),''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual, String message) {
+        return sObjectTypeEquals(actual.getSObjectType(), message);
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType) {
+        return sObjectTypeEquals(objType, expectedMessage(objType,''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType, String message) {
+        return assertSObject(expected.getSObjectType() == objType, message);
+    }
+
+    public ExpectSObject shouldFailToUpsert() {
+        return shouldFailToUpsertWithMessage('Upsert failed');
+    }
+
+    public ExpectSObject shouldFailToUpsertWithMessage(String message) {
```

Comment:
this needs some more work. i would we happy to merge the PR without the shouldFailToUpsert and we tinker with this in a branch. The WithMessage wording has been used to set a custom failure message for the assertion. In the catch block it is used to compare the exception message thrown by upsert with the custom failure message-- which is not the intent of the custom failure message. Line 48 shows this in action with the default failure message 'Upsert failed'. I could be mistaken but that's the way I'm seeing the code right now. 

I feel like it would make sense to have shouldFailToInsert/Update/Delete come in on the same request as shouldFailToUpsert.